### PR TITLE
Main changes in project

### PR DESCRIPTION
**Pull request by Anton-Sikorski**
Creating this PR again due to lack of experience using git

### CODE REVIEW

File: app/controllers/main_controller.rb
Comment (Anton-Sikorski) [inline app/controllers/main_controller.rb:16]: It's temporary method while I'm trying to fix active storage blob load.

File: app/helpers/pokemon_helper.rb
Comment (Anton-Sikorski) [inline app/helpers/pokemon_helper.rb:8]: Here I rewrite Pagy::Frontend method to make navigation look like I want. I'll find way how to avoid this in future.

File: app/javascript/packs/sample.js
Comment (Anton-Sikorski) [inline app/javascript/packs/sample.js:19]: This code is not DRY, but I've decided not to spend lot of time trying to clean it.

File: config/initializers/pagy.rb
Comment (Anton-Sikorski) [inline config/initializers/pagy.rb:2]: I prefer to leave commented hints cause I hope to use them in future

File: spec/controllers/articles_controller_spec.rb
Comment (Anton-Sikorski) [inline spec/controllers/articles_controller_spec.rb]: What else I could test except respond status and template render? Try to imitate user activity with forbidden actions? Should I use Capybara?

File: app/controllers/main_controller.rb
Comment (Anton-Sikorski) [inline app/controllers/main_controller.rb:7]: Here I try to track how attachments load

File: app/models/user.rb
Comment (Anton-Sikorski) [inline app/models/user.rb:20]: And here avatars become attached

File: app/views/comments/_form.html.erb
Comment (Anton-Sikorski) [inline app/views/comments/_form.html.erb:7]: This errors are not intend to show up

File: app/controllers/application_controller.rb
Comment (Anton-Sikorski) [inline app/controllers/application_controller.rb:13]: You asked to describe how authorization works. This is how I understood this task

File: app/javascript/packs/sample.js
Comment (kirylpl) [reply app/javascript/packs/sample.js:19]: That's okay sometimes https://overreacted.io/goodbye-clean-code/

File: app/controllers/main_controller.rb
Comment (kirylpl) [reply app/controllers/main_controller.rb:18]: Please move temp code to the separate branch out of this PR

File: app/helpers/pokemon_helper.rb
Comment (kirylpl) [reply app/helpers/pokemon_helper.rb:8]: Consider using CSS to customize the look. What customization you have performed?

File: spec/controllers/articles_controller_spec.rb
Comment (kirylpl) [reply spec/controllers/articles_controller_spec.rb:42]: > “For new Rails apps: we don’t recommend adding the rails-controller-testing gem to your application. The official recommendation of the Rails team and the RSpec core team is to write request specs instead.”

Comment (kirylpl) [reply spec/controllers/articles_controller_spec.rb:42]: You probably want to check the actual articles creation/edit.
Adding Capybara is good also to test in-browser behavior, let's have one capybara test.

File: app/controllers/main_controller.rb
Comment (kirylpl) [reply app/controllers/main_controller.rb:8]: Consider using sprite image to store all avatars

**Comment by kirylpl** [comment]
Your CI failed, for now, please fix it.
Also please add existing tests to CI.

File: app/views/comments/_form.html.erb
Comment (kirylpl) [reply app/views/comments/_form.html.erb:7]: There should be standard way to display validation errors, please investigate it deeper

**Comment by kirylpl** [comment]
I will not approve this PR because of 5k+  junk files in it.

File: app/assets/stylesheets/pokemon.scss
Comment (kirylpl) [inline app/assets/stylesheets/pokemon.scss:5]: Cool

File: app/controllers/application_controller.rb
Comment (kirylpl) [inline app/controllers/application_controller.rb:21]: Let's have one name for it, either `best_article` or `most_viewed`.
You can left a comment with description.

File: app/controllers/application_controller.rb
Comment (kirylpl) [inline app/controllers/application_controller.rb:42]: Consider using counter cache in the future. (add to TODO, not in this PR)
https://blog.appsignal.com/2018/06/19/activerecords-counter-cache.html
https://guides.rubyonrails.org/association_basics.html#options-for-belongs-to-counter-cache
https://www.dailysmarty.com/posts/how-to-count-page-impressions-in-rails

File: app/controllers/articles_controller.rb
Comment (kirylpl) [inline app/controllers/articles_controller.rb:8]: Good. You can put it in `ApplicationController`.

File: app/controllers/articles_controller.rb
Comment (kirylpl) [inline app/controllers/articles_controller.rb:63]: This should be somewhere else. Please add to the TODO list.

File: app/javascript/packs/sample.js
Comment (kirylpl) [inline app/javascript/packs/sample.js:19]: TODO: add `eslint`

**Comment by kirylpl** [comment]
This is good to go into the main branch. Please resolve conflicts and I'll approve this PR.

**Review by kirylpl — APPROVED** [review]

**Pull request merged**
